File: Backend/src/main/java/com/example/project/dto/request/OrderRequest.java
package com.example.project.dto.request;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.math.BigDecimal;
import java.util.List;


@Data
@AllArgsConstructor
public class OrderRequest {
    @JsonProperty("pickup_point_id")
    @NotNull(message = "ID пункта выдачи не может быть пустым")
    private Integer pickupPointId;

    @JsonProperty("products")
    @Valid
    @NotEmpty(message = "Список продуктов не может быть пустым")
    private List<OrderedProductRequest> orderedProducts;

    @JsonProperty("order_price")
    @NotNull(message = "Сумма заказа не может быть пустой")
    private BigDecimal orderPrice;

}

File: Android/app/src/main/java/com/kire/market_place_android/data/remote/dto/request/order/OrderRequest.kt
package com.kire.market_place_android.data.remote.dto.request.order

import com.kire.market_place_android.data.remote.dto.request.product.OrderedProductRequest
import com.kire.market_place_android.data.remote.dto.response.order.OrderedProductResponse
import com.kire.market_place_android.data.remote.dto.response.util.serializer.BigDecimalSerializer
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.math.BigDecimal

@Serializable
data class OrderRequest(
    @SerialName("pickup_point_id")
    val pickUpPointId: Int = 0,
    @SerialName("products")
    val orderedProducts: List<OrderedProductRequest> = emptyList(),
    @Serializable(with = BigDecimalSerializer::class)
    @SerialName("order_price")
    val orderPrice: BigDecimal = 0.0.toBigDecimal()
)